pkg/kubelet/pleg/generic.go

```diff
@@ -122,34 +165,66 @@ func (g *GenericPLEG) relist() {
 		return
 	}
 
+	// Build a new map.
+	podMap := buildPodMap(pods)
```

Comment:
Yes, I am a little bit concerned about gc, but as the relist frequency is not very high, I guess it would probably be OK. But we can leave a note or todo here so that later it will be helpful if we have some gc issue :)